src/components/atoms/HamburgerMenu/HamburgerMenu.component.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react'
+
+import { Drawer, DrawerProps } from 'antd'
+
+import { PreviewProps } from '@src/types/preview'
+
+import VisibilityComponent from '@components/common/ResponsiveVisibility/ResponsiveVisibility.component'
+
+import { Button, ButtonProps } from '../Button/Button.component'
+import { Icon, IconProps } from '../Icon/Icon.component'
+
+import styles from './HamburgerMenu.module.css'
+
+export enum HamburgerDrawerPlacementEnum {
+  TOP = 'top',
+  RIGHT = 'right',
+  BOTTOM = 'bottom',
+  LEFT = 'left'
+}
+
+export interface HamburgerDrawerProps
+  extends Omit<DrawerProps, 'headerStyle' | 'bodyStyle' | 'footerStyle'> {
+  headerVisible?: boolean
+}
+
+export interface HamburgerMenuProps extends PreviewProps {
+  buttonProps?: Omit<ButtonProps, 'style'>
+  buttonStyle?: ButtonProps['style']
+  drawerProps?: HamburgerDrawerProps
+  headerStyle?: DrawerProps['headerStyle']
+  bodyStyle?: DrawerProps['bodyStyle']
+  iconProps?: IconProps
+  children?: React.ReactNode
+}
+
+const HamburgerMenu = React.forwardRef<HTMLDivElement, HamburgerMenuProps>((props, ref) => {
+  const {
+    isPreview,
+    responsiveVisibility,
+    buttonProps,
+    buttonStyle,
+    drawerProps,
+    headerStyle,
+    bodyStyle,
+    iconProps,
+    children
+  } = props
+  const [previewDrawerVisible, setPreviewDrawerVisible] = useState(false)
+  const { headerVisible, title, closable, ...restDrawerProps } = drawerProps as HamburgerDrawerProps
+
+  const handleButtonClick = () => {
+    setPreviewDrawerVisible(true)
+  }
+
+  const handleDrawerClose = () => {
+    setPreviewDrawerVisible(false)
+  }
+
+  const defaultPreviewButtonProps: ButtonProps = isPreview
+    ? {
+        onClick: handleButtonClick
+      }
+    : {}
+
+  const defaultPreviewDrawerProps: HamburgerDrawerProps = isPreview
```

Comment:
what happen with exported project ? does it work ?

Reply:
@zrg-team Initially Yup it works, I thought the state that hold drawer visibility will be handled by the app, now I move it internally.

https://user-images.githubusercontent.com/20434351/167523831-f84b9505-7ae8-48cd-9252-5b53d702b792.mp4